uniprot-search/src/main/java/org/uniprot/store/search/domain2/UniProtKBSearchFields.java

```diff
@@ -0,0 +1,84 @@
+package org.uniprot.store.search.domain2;
+
+import org.uniprot.core.cv.xdb.UniProtXDbTypes;
+import org.uniprot.store.search.domain2.impl.SearchFieldImpl;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Represents all accessible UniProtKB search fields, and provides access to them via both
+ * contracts, {@link SearchItems} and {@link SearchFields}.
+ *
+ * <p>Created 12/11/2019
+ *
+ * @author Edd
+ */
+public enum UniProtKBSearchFields implements SearchItems, SearchFields {
```

Comment:
Do you think we need SearchItems here? As far as I could understand, I think we will use SearchItems when we load and convert to SearchFields, and also for tests, right? This is why I am not sure if we should also have SearchItems in SearchFields... Does it make sense to you? Sorry, I am not sure if I expressed myself well..

Reply:
makes sense. I'll refactor to make it separate.

The use case for SearchItems is 1.) for the passing the searchitems to a configuration end-point in the configuration controller, for the front-end to use, and 2.) for tests that the examples provided to the front-end are correct.
To achieve 1 and 2, we can load the data via a different path and not expose it in UniProtKBSearchFields.